components/widgets/users/avatar.jsx

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import './avatar.scss';
+
+const Avatar = ({url, username, size}) => (
+    <img
+        className={'Avatar ' + size}
+        alt={`${username || 'user'} profile image`}
+        src={url}
+    />
+);
+
+Avatar.propTypes = {
+    url: PropTypes.string.isRequired,
+    username: PropTypes.string,
+    size: PropTypes.oneOf(['xs', 'sm', 'md', 'lg', 'xl', 'xxl']),
+};
+
+Avatar.defaultProps = {
+    size: 'md',
```

Comment:
Will leave the UX to mike but for the code.
Can we make this `avatar-md` similar to `col-md` or `col-xl`.
I want the sizes to be associated to the component.
Additionally, is there a reason we're going with the capital `A` when beginning the class for the component?

Reply:
This is the approach agreed about how to refer components CSS, you can see aside the avatar.jsx file there is a avatar.scss file which have as root the "Avatar" class, and everything is scoped inside of that.

Is not 100% leak free, somebody can define a "global" class affecting the "xl" class and will affect internally the Avatar component. But the idea is, the size classes here are not affecting anything else besides the avatar. I'm not against adding a prefix to each size class here, but I think it can start to be very uncomfortable if we do it for every single class inside a component. By the way, to maintain this kind of scope/isolation, we should use `Avatar-xl` for example, with the uppercase A to be clear that is a "Component class" which is the policy that we decide to follow.

Reply:
Fixed, see the last commit here: e07611e75bdfcd91765ad63bb7d1b75db827c1c3

Reply:
Thanks.